drivers/spi/spi_nrfx_spim.c

```diff
@@ -178,12 +178,20 @@ static void transfer_next_chunk(struct device *dev)
 		xfer.tx_length   = spi_context_tx_buf_on(ctx) ? chunk_len : 0;
 		xfer.p_rx_buffer = ctx->rx_buf;
 		xfer.rx_length   = spi_context_rx_buf_on(ctx) ? chunk_len : 0;
-		result = nrfx_spim_xfer(&dev_config->spim, &xfer, 0);
-		if (result == NRFX_SUCCESS) {
-			return;
-		}
 
-		error = -EIO;
+#if SOC_NRF52832 /* This SPIM driver is only used by the NRF52832 if SOC_NRF52832_PAN_58_OVERRIDE is enabled */
```

Comment:
Outside of Kconfig files, the options defined there appear with their names prefixed with `CONFIG_`. So you need to use `CONFIG_SOC_NRF52832` here (and since the macro may be not defined, it would be better to use `#ifdef`).
Same applies to the change in `CMakeLists.txt`.

Reply:
you can also avoid #ifdefs and do something like:
```
if (IS_ENABLED(CONFIG_SOC_NRF52832) &&
    (xfer.rx_length == 1 && xfer.tx_length <= 1)) {
			LOG_DBG("Transaction aborted since it would trigger NRF52832 PAN 58");
			error = -EIO;
}
```

Reply:
@anangl and @nordic-krch - thank you for taking the time to catch such a "newbie" error before I could see the results in CI.

Honestly - I've been doing too much "enterprise Java" (day job) and it's been too long since I've played with bare metal...

Done.